src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -182,18 +182,25 @@ data SnapLevel = SnapLevel {
   deriving stock (Show, Eq, Read)
 
 data SnapMergingRun =
-    SnapMergingRun !MergePolicyForLevel !NumRuns !SnapMergingRunState
+    SnapMergingRun !MergePolicyForLevel !NumRuns !NumEntries !UnspentCredits !MergeKnownCompleted !SnapMergingRunState
```

Comment:
Do we need to snapshot the `MergeKnownCompleted`? It's like a cache, we can just reconstruct it from the `SnapMergingRunState`.

Reply:
I've thought about this as well, and I think both with and without storing it here works. It just means we can skip a bit of work

Reply:
I think I'd rather keep the snapshot format lean. This would just mean two more lines of code in opening a snapshot and even save one when creating it I think, or are you referring to some other work?